### DISCUSSION TITLE
Subscribe example typings don't seem to be correct

### DISCUSSION THREAD

**Discussion by chrissysemens** [pmndrs/zustand#1858] [Bug report] [answered]
In attempting to subscribe to a single piece of store data as per this example:
![01](https://github.com/pmndrs/zustand/assets/46443589/aeabeb06-65f9-4339-b1b8-5b142b88eb6f)

I get type erorrs:
![02](https://github.com/pmndrs/zustand/assets/46443589/223fa004-978e-47f1-9d31-005d772c3662)

Expects 1 param and not 2.
![001](https://github.com/pmndrs/zustand/assets/46443589/f2cb9483-edf8-4550-9f12-8d59fa1a9480)

Has something changed? Is there a working example?

**Comment by samhirtarif** [comment]
What version of zustand are you using?

**Reply by chrissysemens** [reply]
Thanks man! Appreciated. 

^4.3.8

**Reply by samhirtarif** [reply, marked as answer]
~~I don't think that version supports a second argument for the subscribe method. I am using the same version in one of my projects and it seems that it only takes one method as a parameter.~~

~~Seems like the docs may be out-dated 🤷🏻‍♂️~~

Actually no, the docs are right, I just missed a point. So as the docs [here](https://docs.pmnd.rs/zustand/recipes/recipes#reading/writing-state-and-reacting-to-changes-outside-of-components) state, if we want to use the second parameter on the `subscriber` method, we need to have the `subscribeWithSelector` middleware on our store. Are you sure you have that?

**Reply by chrissysemens** [reply]
Totally what I'd missed. Thank you!
